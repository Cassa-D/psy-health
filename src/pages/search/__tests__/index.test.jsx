import { mount } from 'enzyme'
import React from 'react'

import Search from '../index'
import SearchFilter from '../../../containers/search-filter'
import { useDispatch } from 'react-redux';
import PsychologistList from '../../../components/psychologist-list';


jest.mock("react", () => ({
  // mock do useState
  ...jest.requireActual("react"),
}));

jest.mock("react-redux", () => ({
  // mock do useDispatch e do useSelector
  useDispatch: jest.fn(),
  useSelector: jest.fn(() => []),
}));


describe('Renders', () => {

  const dispatchMock = jest.fn()
  const useDispatchMock = () =>  dispatchMock

  useDispatch.mockImplementation(useDispatchMock)

  const wrapper = mount(<Search />)

  it ('Snapshot', () => {
    
    expect(wrapper).toMatchSnapshot()
  })


  it ('toBeTruthy SearchFilter', () => {
    expect(wrapper.find(<SearchFilter />)).toBeTruthy()
  })

  it('toBeTruthy psychologist list ', () => {
    expect(wrapper.find(<PsychologistList psychologists={[{ name: "joao" }, { name: "pedro" }]}/>)).toBeTruthy()
  })

  it('call', () => {
    expect(dispatchMock).toBeCalledTimes(3)
    expect(dispatchMock).toBeCalledWith({ filtered: [], type: "FILTERED_PSY" })
  })
 
})