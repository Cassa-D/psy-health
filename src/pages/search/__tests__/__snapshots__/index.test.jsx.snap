// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Snapshot 1`] = `ReactWrapper {}`;
